pennylane_ionq/device.py

```diff
@@ -319,30 +332,286 @@ def _apply_operation(self, operation, circuit_index=0):
-        par = operation.parameters
-
-        if len(wires) == 2:
-            if name in {"SWAP", "XX", "YY", "ZZ", "MS"}:
-                # these gates takes two targets
-                gate["targets"] = wires
-            else:
-                gate["control"] = wires[0]
-                gate["target"] = wires[1]
+        params = operation.parameters
+        if name == "Evolution":
+            num_trotter_steps = (
+                operation.num_steps if operation.num_steps is not None else 1
+            )
+            terms = self._extract_evolution_pauli_terms(operation, wires)
+            coefficients = self._extract_evolution_coefficients(operation, wires)
+            terms, coefficients = self.remove_trivial_terms(terms, coefficients)
+            if len(terms) > 0:
+                for _ in range(num_trotter_steps):
+                    gate = {"gate": self._operation_map[name]}
+                    gate["targets"] = wires
+                    gate["terms"] = terms
+                    gate["coefficients"] = [-1 * float(v) for v in coefficients]
+                    gate["time"] = operation.param / num_trotter_steps
+                    self.input["circuits"][circuit_index]["circuit"].append(gate)
+        elif name == "ParametrizedEvolution":
+            terms = self._extract_parametrized_evolution_pauli_terms(operation, wires)
+            coefficients = self._extract_parametrized_evolution_coefficients(
+                operation, wires
+            )
+            terms, coefficients = self.remove_trivial_terms(terms, coefficients)
+            if len(terms) > 0:
+                previous_time = operation.t[0]
+                time_steps = operation.t.tolist()[1:]
+                for time_step in time_steps:
+                    time_delta = time_step - previous_time
+                    parameters = operation.parameters
+                    # TODO: @CODE REVIEWER
+                    # (1) at this point I am not anymore convinced we should provide an implementation for ParametrizedEvolution operator here
+                    # (2) below we assume that we the array of time points provided by user specify the intermediat points where the
+                    #     hamiltonian should be evaluated. I am not sure this assumption is correct. Also note that I am evaluting the coefficient
+                    #     in the middle of the time interval. This intuitevly makes sense, but I am not sure this is what we should be doing.
+                    # (3) Here I assume the hamiltonian depends on some  unspecified parametrs and one more argument wich is time. I do not know
+                    #     how to enforce this choice in API as the user may come up with a different parametrization, live having time as the first 
+                    #     argument for example.
+                    # (4) It is not clear how the number of steps for trotterization should be provided as input.

```

Comment:
`ParametrizedEvolution` is not decomposed by Trotterization. In a simulation setting, we use an ODE solver to time-evolve the state vector based on the time steps. There is no way to specify the number of Trotter steps when using a `ParametrizedEvolution`, so the way to apply it to a QPU is not immediately clear to me. I will get input from the team to give you a thorough response.